internal/resourcestore/resourcestore.go

```diff
@@ -53,12 +60,17 @@ func New() *ResourceStore {
 func NewWithTimeout(timeout time.Duration) *ResourceStore {
 	rc := &ResourceStore{
 		resources: make(map[string]*Resource),
+		closeChan: make(chan struct{}, 1),
 		timeout:   timeout,
 	}
 	go rc.cleanupStaleResources()
 	return rc
 }
 
+func (rc *ResourceStore) Close() {
+	rc.closeChan <- struct{}{}
+}
```

Comment:
should it also do `close(rc.closeChan)`?

Reply:
good point, it should actually only close methinks, as the receiever will still get a value out of it and know it's time to return